test-case/src/ContaoTestCase.php

```diff
@@ -133,10 +133,15 @@ protected function mockContaoFramework(array $adapters = [], array $instances =
             ->willReturnCallback(static fn (string $key): Adapter|null => $adapters[$key] ?? null)
         ;
 
+        $instances = array_map(
+            static fn ($instance): \Closure => $instance instanceof \Closure ? $instance : static fn () => $instance,
+            $instances,
+        );
+
         if ($instances) {
             $framework
                 ->method('createInstance')
-                ->willReturnCallback(static fn (string $key): mixed => $instances[$key] ?? null)
+                ->willReturnCallback(static fn (string $key): mixed => ($instances[$key] ?? null)?->__invoke())
```

Comment:
```suggestion
                ->willReturnCallback(
                    static function (string $key): mixed {
                        if (!isset($instances[$key])) {
                            return null;
                        }

                        if ($instances[$key] instanceof \Closure) {
                            return $instances[$key]();
                        }
                        
                        return $instances[$key];
                    }
                )
```

This is better understandable and does not require to convert all instances to closures in lines 136–139.

Reply:
Changed in 9d625d5ab5c82c5bb1b9e24497d399abd86f19c9.